components/list/list-item-drag-mixin.js

```diff
@@ -33,86 +94,286 @@ export const ListItemDragMixin = superclass => class extends superclass {
+		const dropSpots = dropSpotsFactory();
+		if (this === dropSpots.target) {
+			return;
+		}
+		dropSpots.addVisitor(this);
+		this._dropTarget = true;
+		e.dataTransfer.dropEffect = 'move';
 	}
 
-	// TODO
-	_removeCopy() {
+	_renderBottomPlacementMarker(renderTemplate) {
+		return this._bottomPlacementMarker ? html`<div class="d2l-list-item-drag-bottom-marker">${renderTemplate}</div>` : null;
+	}
 
+	_renderDraggableArea(templateMethod) {
+		templateMethod = templateMethod || (dragArea => dragArea);
+		return this.draggable && !this._keyboardMode ? templateMethod(html`
+			<div
+				class="d2l-list-item-drag-area"
+				draggable="true"
+				@click="${this._onDragAreaClick}"
+				@dragstart="${this._onDragStart}"
+				@dragend="${this._onDragEnd}"
+				>
+			</div>
+		`) : nothing;
 	}
 
-	_renderDragAction(inner) {
-		return this.draggable ? html`
-			<div @click="${this._handleDragActionClick}" class="d2l-list-item-drag-action">${inner}</div>
-			` : nothing;
+	_renderDragHandle(templateMethod) {
+		templateMethod = templateMethod || (dragHandle => dragHandle);
+		return this.draggable ? templateMethod(html`
+			<d2l-list-item-drag-handle id="${this._itemDragId}" @d2l-list-item-drag-handle-action="${this._onDragHandleActions}"></d2l-list-item-drag-handle>
+		`) : nothing;
 	}
-	_renderDragHandle() {
-		return this.draggable ? html`
-			<div
-				id="${this._itemDragId}"
-				class="d2l-list-item-drag"
-				@dragenter="${this._dragEnter}"
-				@dragleave="${this._dragExit}"
-			></div>
-		` : nothing;
+
+	_renderDropArea(templateMethod) {
+		templateMethod = templateMethod || (dropArea => dropArea);
+		return this.draggable && this._dropTarget ? templateMethod(html`
+			<div class="d2l-list-item-drag-drop-grid" @drop="${this._onDrop}" @dragover="${this._onDragOver}">
+				<div @dragenter="${this._onDragTopEnter}"></div>
+				<div @dragenter="${this._onDragUpperEnter}"></div>
+				<div @dragenter="${this._onDragLowerEnter}"></div>
+				<div @dragenter="${this._onDragBottomEnter}"></div>
+			</div>
+		`) : nothing;
+	}
+
+	_renderTopPlacementMarker(renderTemplate) {
+		return this._topPlacementMarker ? html`<div class="d2l-list-item-drag-top-marker">${renderTemplate}</div>` : null;
 	}
 };
 
+let dropSpots = null;
+
+function dropSpotsFactory(target) {
+	if (!dropSpots) dropSpots = new DropSpotsState(target);
+	return dropSpots;
+
+}
+
+function dropSpotsBlowUp() {
+	if (dropSpots) {
+		dropSpots.clear();
+	}
+	dropSpots = null;
+}
+
+class DropSpotsState {
+	constructor(target) {
+		this._target = target;
+		this._destination = null;
+		this._visitedDestination = new Map();
+		this._insertBefore = false;
+		this._time = 0;
+	}
+
+	addVisitor(destination) {
+		if (destination && !this._visitedDestination.has(destination)) {
+			this._visitedDestination.set(destination, null);
+		}
+	}
+
+	clear() {
+		this._cleanUpOnLeave();
+		this._visitedDestination.forEach((_, visitedDestination) => visitedDestination._dropTarget = false);
```

Comment:
Would be nice to avoid having to do this (per previous comment).

Reply:
As explained not possible :(